src/main/java/com/autotune/operator/InitializeDeployment.java

```diff
@@ -15,66 +15,131 @@
-		if (monitoring_agent_endpoint == null || monitoring_agent_endpoint.isEmpty()) {
-			if (monitoring_agent == null || monitoring_agent_service == null) {
-				throw new MonitoringAgentNotFoundException();
-			} else {
-				// Fetch endpoint from service cluster IP
-				monitoring_agent_endpoint = DataSourceFactory.getDataSource(monitoring_agent).getDataSourceURL();
-			}
-		}
-		KruizeDeploymentInfo.setMonitoringAgentEndpoint(monitoring_agent_endpoint);
 
-		KruizeDeploymentInfo.setLayerTable();
+    }
 
-		KruizeDeploymentInfo.initiateEventLogging();
+    private static void setConfigValues(String configFileName, Class envClass) {
+        String configFile = System.getenv(configFileName);
+        JSONObject configObject = null;
+        if (null != configFile) {
+            try (InputStream is = new FileInputStream(configFile)) {
+                String jsonTxt = new String(is.readAllBytes(), StandardCharsets.UTF_8);
+                configObject = new JSONObject(jsonTxt);
+            } catch (Exception exception) {
+                LOGGER.warn("Failed to set config using file {} due to {}. checking if corresponding environment variable is set.", configFile, exception.getMessage());
+            }
+        }
+        List<String> systemENVList = new ArrayList<>();
+        Field[] fields = envClass.getFields();
+        for (Field field : fields) {
+            if (field.getType() == String.class) {
+                try {
+                    KruizeDeploymentInfo.class.getDeclaredField(field.getName()).set(null,
+                            getKruizeConfigValue((String) field.get(null), configObject));
+                } catch (IllegalAccessException | NoSuchFieldException e) {
+                    LOGGER.warn("Error while setting config variables :  {}", e.getMessage());
+                }
+            }
+        }
+    }
 
-		KruizeDeploymentInfo.logDeploymentInfo();
+    private static String getKruizeConfigValue(String envName, JSONObject kruizeConfigObject) {
+        Object envValue = null;
+        String message = "Config variable : " + envName;
+        try {
+            if (null != kruizeConfigObject) {
+                //extract values from jsonObject
+                try {
+                    String[] parts = envName.split("_");
+                    JSONObject primaryObj = kruizeConfigObject;
+                    int i = 0;
+                    for (i = 0; i < parts.length - 1; i++) {
+                        primaryObj = primaryObj.getJSONObject(parts[i]);
+                    }
+                    envValue = primaryObj.get(parts[i]);
+                    message = message + "  :- found in mounted config file";
+                } catch (Exception e) {
+                    message = message + "  :- Not found in mounted config file";
+                }
+            } else {
+                message = message + "  :- No mount config file";
+            }
+            //Override if env value set outside kruizeConfigJson
```

Comment:
Has this been tested? (ENV vars getting precedence over config JSON?)

Reply:
Certainly. Kruize follows a specific logic when it comes to configuration variables. If a configuration variable is set but not defined in the environment variables, Kruize will use the configuration variables. If a configuration variable is not set, but it is defined in the environment variables, Kruize will use the environment variables. However, if both configuration variables and environment variables are set, Kruize will prioritize the environment variables and override the configuration variables.